components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -163,4 +179,31 @@ mod tests {
             assert_eq!(output, None, "{:?} {:?}", date, format);
         }
     }
+
+    #[test]
+    fn test_week_day() {
+        let cases = vec![
+            ("2018-12-03", 0i64),
+            ("2018-12-04", 1i64),
+            ("2018-12-05", 2i64),
+            ("2018-12-06", 3i64),
+            ("2018-12-07", 4i64),
+            ("2018-12-08", 5i64),
+            ("2018-12-09", 6i64),
+        ];
```

Comment:
Invalid case should be tested as well.

Reply:
@sticnarf Thanks, but I'm not sure if I understand what you mean by "invalid case". If you mean that the input string has an invalid date format, then it can be handled outside our vectorized function. If you mean that `Option<DateTime>` might be a `None`, this case has already been added below.

Reply:
Sorry I didn't state it clearly. I mean the `if t.is_zero()` branch. Test cases only contain valid dates and null, but a date containing zero isn't included.

Reply:
Thanks, test cases updated.